### PR TITLE
CI: Add a MetalLB E2E lane

### PR DESCRIPTION
**Pull request by oribon**
We add a MetalLB lane to make sure we aren't introducing changes that break the integration between the two projects.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -187,3 +187,92 @@ jobs:
...
+          mkdir -p /tmp/kind_logs/
+
+      - name: Checkout MetalLB
+        uses: actions/checkout@v4
+        with:
+          repository: metallb/metallb
+          path: metallb
+          ref: "main"
+
+      - name: Deploy MetalLB
```
Comment (fedepaol) [inline .github/workflows/ci.yaml:238]: Why not just deploying metallb in frr mode overriding the frrk8s image? What am I missing?

Comment (oribon) [reply .github/workflows/ci.yaml:238]: I think it would be messy when prs change the helm chart itself (calling the new changes from metallb's chart)

Comment (fedepaol) [reply .github/workflows/ci.yaml:238]: I _think_ we could just copy the whole chart folder under metallb's one (which is what we end up doing when we declare frrk8s as a metallb dependency anyway). That'd remove all the logic we are adding here.
Does it make sense? I think it's worth an attempt

Comment (oribon) [reply .github/workflows/ci.yaml:238]: sure I can try

Comment (oribon) [reply .github/workflows/ci.yaml:246]: it looks like I made it work, but not sure I like the outcome (the one later where we will use `inv dev-env`) 😅 seems a bit hacky (unless that's not what you meant), and will also require updating dev-env to accept extra helm args + loading more images into kind. wdyt should we continue with this or revert to what I originally did?

Comment (fedepaol) [reply .github/workflows/ci.yaml:246]: yep let's revert and see if this won't bite us in the future :P

Comment (oribon) [reply .github/workflows/ci.yaml:238]: done

**Pull request merged**